### PR TITLE
[Enhancement] Persist sidebar visibility between sessions

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/16006

This PR adds a local storage a setting for whether the app side bar is open or closed and uses it to update the app to reflect that value. CoreUI controls the side bar with a class added to or removed from the `body` tag of `index.html`. This falls outside of the DOM node that the React app is appended to so we can't use [React refs](https://react.dev/reference/react/useRef#manipulating-the-dom-with-a-ref) here. But, we can use the [MutationObserver API](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) and query selectors to reach outside the app and watch the class changes.

## Testing
**URL to test:** 
https://deploy-preview-1424--atd-vze-staging.netlify.app/

**Steps to test:**
1. Go to any view of the app
2. Check your local storage for a key called `isAppSideBarOpen`. On the first time this key is initialized, it should have a value of `true` and the sidebar should be open.
3. Refresh the page, the local storage value should remain `true`.
4. Now click on the hamburger icon on the side bar header to close the side bar. The local storage value should now be `false`.
5. Refresh the page or close the app and come back. When the app loads, it should load with the side bar closed.
6. Try updating the value of `isAppSideBarOpen` in your browser settings manually to either `true` or `false` and refresh. The app will use this value to either hide or show the side bar on load.
7. Try out different combinations, and your local storage value should always match the status of the side bar.

---
#### Ship list
- [x] Check migrations for any conflicts with latest migrations in master branch
- [x] Confirm Hasura role permissions for necessary access
- [x] Code reviewed
- [x] Product manager approved

### CODE REVIEW

**Review by johnclary — APPROVED** [review]
this worked for me at first, and then when i opened my debug console it stopped working, and so i debugged it some more, and then i finally realized that when my debug console is open, the `sidebar-lg-show` class is not active because the sidebar is autohiding. d'oh!


this is fantastic. and TIL about `MutationObserer`s. this is a really nice fix, @mddilley! thank you! 🚢 🚢 🚢 🚢 🚢

File: atd-vze/src/containers/DefaultLayout/DefaultLayout.js [hunk 1]
```diff
@@ -25,10 +25,69 @@ const DefaultAside = React.lazy(() => import("./DefaultAside"));
...
+  const [isAppSideBarOpen, setIsAppSideBarOpen] = React.useState(
+    doesLocalStorageExist ? JSON.parse(initialIsAppSideBarOpen) : true
+  );
+
+  // On mount, set an observer that watches the body element for changes in the class attribute
+  // and updates the state and local storage to match the classes used by CoreUI to show/hide the sidebar
+  React.useEffect(() => {
+    const elemToObserve = document.getElementById(appBodyId);
+
+    const observer = new MutationObserver(function(mutations) {
```
Comment (roseeichelmann) [inline atd-vze/src/containers/DefaultLayout/DefaultLayout.js:50]: this is super cool! big TIL for me :)

**Review by roseeichelmann — APPROVED** [review]
Works like a charm, this was such a cool solution mike and nicely executed :raised_hands:

**Comment by mddilley** [comment]
thanks y'all! Mergin' 🚢

**Pull request merged**